app/screens/settings/display_settings/display_settings.js

```diff
@@ -32,9 +34,12 @@ export default class DisplaySettings extends PureComponent {
         showClockDisplaySettings: false,
     };
 
-    constructor(props) {
-        super(props);
-        props.navigator.setOnNavigatorEvent(this.onNavigatorEvent);
+    componentDidMount() {
+        this.navigationEventListener = Navigation.events().bindComponent(this);
+    }
+
+    componentDidAppear() {
+        setNavigatorStyles(this.props.componentId, this.props.theme);
```

Comment:
Perhaps this should be done in `componentDidMount`? We only need to use `didAppear` and `didDisappear` if we're having other screens pushed on top of this.